### PR TITLE
For #633 - Gives the user the ability to save their session

### PR DESCRIPTION
**Pull request by boek**

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -42,6 +43,12 @@ import org.mozilla.fenix.mvi.getAutoDisposeObservable
 import org.mozilla.fenix.mvi.getManagedEmitter
 import kotlin.math.roundToInt
 
+fun SessionBundleStorage.archive(sessionManager: SessionManager) {
+        save(sessionManager.createSnapshot())
+        sessionManager.removeAll()
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Do we also want to remove custom tab sessions? That's what removeAll() does.

File: app/src/main/res/layout/component_tabs.xml [hunk 2]
```diff
@@ -1,8 +1,82 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.recyclerview.widget.RecyclerView
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<merge
```
Comment (colintheshots) [inline app/src/main/res/layout/component_tabs.xml]: Nice. It's a good idea to use merge layouts.

File: app/src/main/res/layout/component_tabs.xml [hunk 3]
```diff
@@ -1,8 +1,82 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/tabs_header_title"
+            android:textAppearance="@style/TextAppearance.MaterialComponents.Headline6"
+            android:textColor="?attr/toolbarTextColor"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <ImageButton
```
Comment (colintheshots) [inline app/src/main/res/layout/component_tabs.xml]: The ImageButtons should have contentDescriptions for A11y.

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt (hunk 1 above)
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: good point, also private tabs

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 4]
```diff
@@ -42,6 +43,14 @@ import org.mozilla.fenix.mvi.getAutoDisposeObservable
 import org.mozilla.fenix.mvi.getManagedEmitter
 import kotlin.math.roundToInt
 
+fun SessionBundleStorage.archive(sessionManager: SessionManager) {
+        save(sessionManager.createSnapshot())
+        sessionManager.sessions.filter { !it.private }.forEach {
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Do we allow archiving private sessions? I'll assume not for the moment.

Comment (boek) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: @colintheshots Negative. The session storage filters out all private sessions before archiving.

I figured for now we would keep your private sessions around if there is one in memory.

**Pull request merged**